File: ged-backend/src/main/java/com/loga/skeleton/wrapper/ContenuDossierWrapper.java
package com.loga.skeleton.wrapper;

import com.loga.skeleton.domain.entity.Document;
import com.loga.skeleton.domain.entity.Dossier;
import com.loga.skeleton.domain.entity.Raccourci;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class ContenuDossierWrapper {
    List<Document> documents = new ArrayList<>();
    List<Dossier> dossiers  = new ArrayList<>();
    Dossier dossier;
    List<Raccourci> raccourcis  = new ArrayList<>();

    public static ContenuDossierWrapper of( List<Document> documents, List<Dossier> dossiers,List<Raccourci> raccourcis ,Dossier dossier){
        return new ContenuDossierWrapper(documents, dossiers, dossier, raccourcis);
    }
}

File: ged-web-client/src/app/main/skeleton/configuration/wrapper/contenu-dossier-wrapper.ts
import {Dossier, Raccourci} from "../_model";
import {Document} from "../_model";

export class ContenuDossierWrapper  {

    public documents: Document[] = [];
    public dossiers: Dossier[]= [];
    public raccourcis: Raccourci[] = [];
    public dossier: Dossier = new  Dossier();
}
